### PR TITLE
CPM-74: add private views

### PR DESCRIPTION
**Pull request by nmarniesse**
**Description (for Contributor and Core Developer)**

Back:
- Add integration tests for the controller (only save)
- Move integration tests for the repository and add some tests in it
- Update repository to fetch only authorized views: owned views + public views
- Add a check in save: cannot edit a view if the alias is not the good one (before we just load the view by the id)
- Add a check in save: cannot edit a view if the user is not the owner
- Once the view is created, the type (public or private) cannot be changed

Front:
- Add "personal view" checkbox in the create form
- Add the "public" label in view list when the view is public

create_private_view

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | OK
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Translator/TranslatorDecorator.php [hunk 1]
```diff
@@ -13,7 +14,7 @@
  * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-class TranslatorDecorator implements TranslatorInterface
+class TranslatorDecorator implements LegacyTranslatorInterface, TranslatorInterface
```
Comment (nmarniesse) [inline src/Akeneo/Platform/Bundle/UIBundle/Translator/TranslatorDecorator.php:17]: LegacyTranslatorInterface is deprecated and will be removed in Symfony 5. Replaced by TranslatorInterface here.  
But we still need to implements LegacyTranslatorInterface in order to not break the code.

File: src/Oro/Bundle/PimDataGridBundle/Repository/DatagridViewRepository.php [hunk 2]
```diff
@@ -23,26 +23,13 @@ public function getDatagridViewTypeByUser(UserInterface $user)
...
+                ->setParameter('type', DatagridView::TYPE_PUBLIC)
+                ->setParameter('owner', $user)
             ->getQuery()
             ->execute();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function findDatagridViewByAlias($alias)
```
Comment (nmarniesse) [inline src/Oro/Bundle/PimDataGridBundle/Repository/DatagridViewRepository.php:33]: Method is deprecated in the interface, and it is never used

File: src/Oro/Bundle/PimDataGridBundle/tests/Integration/Repository/DatagridViewRepositoryIntegration.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+use Akeneo\UserManagement\Component\Model\UserInterface;
+use Akeneo\UserManagement\Component\Repository\UserRepositoryInterface;
+use Oro\Bundle\PimDataGridBundle\Entity\DatagridView;
+use Oro\Bundle\PimDataGridBundle\Repository\DatagridViewRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Pierre Allard <pierre.allard@akeneo.com>
+ */
+class DatagridViewRepositoryIntegration extends TestCase
```
Comment (nmarniesse) [inline src/Oro/Bundle/PimDataGridBundle/tests/Integration/Repository/DatagridViewRepositoryIntegration.php:17]: This class is moved (github does not follow it) and some tests are updated/added

File: src/Oro/Bundle/PimDataGridBundle/Repository/DatagridViewRepositoryInterface.php [hunk 4]
```diff
@@ -23,18 +23,6 @@ interface DatagridViewRepositoryInterface
...
-     * Get all datagrid views by type
-     *
-     * @param UserInterface $user
-     * @param string        $alias
-     *
-     * @return ArrayCollection
-     *
-     * @deprecated Please use DatagridViewRepositoryInterface::findDatagridViewBySearch instead
-     */
-    public function findDatagridViewByAlias($alias);
```
Comment (mmetayer) [inline src/Oro/Bundle/PimDataGridBundle/Repository/DatagridViewRepositoryInterface.php:36]: please add an entry in the CHANGELOG's BC breaks section

**Review by mmetayer — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/grid/view-selector-create-view.js [hunk 5]
```diff
@@ -77,16 +78,14 @@ define([
...
-            label: __('pim_datagrid.view_selector.choose_label'),
-          }),
+          fields: this.templateInput({ __ }),
         }),
       });
 
       modal.open();
 
       const $submitButton = modal.$el.find('.ok').addClass('AknButton--disabled');
+      const $typeSelector = modal.$el.find('#new-view-type-switcher');
```
Comment (pierallard) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/grid/view-selector-create-view.js:88]: We try to not use identifiers. Please use something else (a class, for example)

File: src/Oro/Bundle/PimDataGridBundle/tests/Integration/Controller/Rest/CreateOrUpdateDatagridViewIntegration.php [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        $datagridView->setOwner($this->loggedUser);
+        $datagridView->setColumns(['identifier']);
+        $datagridView->setFilters('filters');
+        $this->datagridViewSaver->save($datagridView);
+        $this->fixtureViewIds['admin_view'] = $datagridView->getId();
+
+        $datagridView = new DatagridView();
+        $datagridView->setDatagridAlias('product-grid');
+        $datagridView->setLabel('Mary\'s view');
+        $datagridView->setType(DatagridView::TYPE_PUBLIC);
```
Comment (pierallard) [inline src/Oro/Bundle/PimDataGridBundle/tests/Integration/Controller/Rest/CreateOrUpdateDatagridViewIntegration.php:199]: Even if there is no tests which is testing this, I would set one view to PRIVATE, to ensure "more code coverage"

File: src/Oro/Bundle/PimDataGridBundle/tests/Integration/Repository/DatagridViewRepositoryIntegration.php (hunk 3 above)
Comment (pierallard) [reply src/Oro/Bundle/PimDataGridBundle/tests/Integration/Repository/DatagridViewRepositoryIntegration.php:17]: Do I really code that class ? :shrug:

**Review by pierallard — APPROVED** [review]

**Pull request merged**